client/post-editor/editor-revisions-list/index.jsx

```diff
@@ -178,22 +185,28 @@ class EditorRevisionsList extends PureComponent {
 	}
 }
 
+const filterNewRevisionsIds = ( newRevisionIds, revisions ) =>
+	reject( newRevisionIds, x => includes( map( revisions, 'id' ), x ) );
+
 export default connect(
-	( state, { comparisons, revisions, selectedRevisionId } ) => {
+	( state, { comparisons, revisions, selectedRevisionId, postId } ) => {
 		const { nextRevisionId, prevRevisionId } = get( comparisons, [ selectedRevisionId ], {} );
 		const latestRevisionId = get( head( revisions ), 'id' );
 		const latestRevisionIsSelected = latestRevisionId === selectedRevisionId;
 		const earliestRevisionIsSelected =
 			! latestRevisionIsSelected && get( last( revisions ), 'id' ) === selectedRevisionId;
 		const nextIsDisabled = latestRevisionIsSelected || revisions.length === 1;
 		const prevIsDisabled = earliestRevisionIsSelected || revisions.length === 1;
+		const revisionIds = PostEditStore.getRevisionIds( postId );
```

Comment:
Why aren't we getting these out of the global state tree via [`getPostRevisions`](https://github.com/Automattic/wp-calypso/blob/23ea34b7b1847e7dd6290453bbcc110acdfc9388/client/state/selectors/get-post-revisions.js#L14) instead of maintaining more state in `PostEditStore`?

Reply:
Hm! Thanks Jeff, I'll have a look at that.

Reply:
If there are incoming new revisions from the post editor, the `PostEditStore` gives them to the `EditorRevisionsList` without delay. When you view the history, make changes and reopen the history, the `PostEditStore` list is available before the component's mounted, whereas a `getPostRevisions` call and the `revisions` prop only get the updated list three "receive props" cycles later, just before the new items are rendered.

The first way, a placeholder warns us something's coming; the second way, the placeholder appears for such a brief time it's hardly visible. My guess is it's probably not going to work well enough that way.